web/static/js/components/timezones/TimezoneAutocomplete.jsx

```diff
@@ -0,0 +1,103 @@
+import React, { PropTypes, Component } from 'react'
+import { fetchTimezones } from '../../actions/timezones'
+import { connect } from 'react-redux'
+import { translate } from 'react-i18next'
+import { Autocomplete } from '../ui'
+import * as actions from '../../actions/survey'
+import { formatTimezone } from './util'
+import 'materialize-autocomplete'
+
+class TimezoneAutocomplete extends Component {
+  static propTypes = {
+    t: PropTypes.func,
+    dispatch: PropTypes.func.isRequired,
+    selectedTz: PropTypes.string,
+    timezones: PropTypes.object
+  }
+
+  constructor(props) {
+    super(props)
+    this.timeZoneInput = null
+    this.state = {
+      fetching: false
+    }
+  }
+
+  setTimezoneInputVal = (timezone) => {
+    if (this.timeZoneInput) this.timeZoneInput.value = timezone
+  }
+
+  focusTimezoneInput = () => {
+    if (this.timeZoneInput) this.timeZoneInput.focus()
+  }
+
+  getTimezoneInputRef = () => {
+    return this.timeZoneInput
+  }
+
+  componentDidUpdate() {
+    const { selectedTz } = this.props
+    this.focusTimezoneInput()
+    this.setTimezoneInputVal(formatTimezone(selectedTz))
+  }
+
+  componentDidMount() {
+    const { dispatch, selectedTz } = this.props
+    dispatch(actions.setTimezone(selectedTz))
+    this.focusTimezoneInput()
+    this.setTimezoneInputVal(formatTimezone(selectedTz))
+    dispatch(fetchTimezones())
+  }
+
+  buildTimezonesforAutoselect(options) {
+    return options.map((element) => {
+      return { text: element, id: element }
```

Comment:
Shouldn't we `formatTimezone` the option that's shown on the dropdown while performing the autoselection?

With the code as-is, while you're typing `phno` you see an option like `Asia/Phnom_Penh`, and when you confirm it you see `Asia - Phonm Penh`.